docs/source/gravlensingintro.ipynb

```diff
@@ -0,0 +1,1538 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load_ext autoreload\n",
+    "%autoreload 2\n",
+    "\n",
+    "import torch\n",
+    "import matplotlib.pyplot as plt\n",
+    "import matplotlib.animation as animation\n",
+    "from matplotlib.patches import Circle\n",
+    "import matplotlib\n",
+    "from IPython.display import HTML\n",
+    "import numpy as np\n",
+    "\n",
+    "import caustics\n",
+    "from caustics.constants import arcsec_to_rad"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# setup stuff to define a grid of points to make images\n",
+    "\n",
+    "n_pix = 256\n",
+    "res = 0.015\n",
+    "fov = n_pix * res\n",
+    "upsample_factor = 2\n",
+    "z_l = torch.tensor(0.5, dtype=torch.float32)\n",
+    "z_s = torch.tensor(1.0, dtype=torch.float32)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Lensing Formalism\n",
+    "\n",
+    "### Configuration of a Lens\n",
+    "\n",
+    "Before we get too deep into the weeds, lets set the stage. A photon has arrived in your telescope and you are wondering where it came from. Well if it encountered some mass on its path to you (and really what photon hasn't?) then the path it took might look something like this:\n",
+    "\n",
+    "![Lens configuration](assets/lensconfiguration.png)\n",
+    "\n",
+    "*Figure taken from Bartelmann and Schneider 2001 (figure 11)*\n",
+    "\n",
+    "The center of your image is your (0,0) coordinate and this corresponds to the *optical axis* which is the dashed line. The source of the photon is said to exist in the *source plane* which is a distance of $D_s$ from the observer. The mass that deflected the photon is said to exist in the *lens plane* which is a distance $D_l$ (I use $D_l$ rather that $D_d$ for the sake of my sanity) from the observer and $D_{ls}$ from the source plane. Finally, you receive the photons in the *image plane* at the observer. Note that cosmological distances are so large that these may be very accurately treated as infinitely thin planes, even if the source and the mass are in fact 3D structures. We'll talk a little about extended mass distributions in the [Multiplane Lensing](#multiplane-lensing) section. In the source plane the photons are coming from a point $\\vec{\\eta}$ away from the optical axis, this is a physical distance you could measure in kilometers if you like. In the lens plane, the photon intersected at $\\vec{\\xi}$ away from the optical axis (again physical units). In the image plane we can say the photon was essentially at the optical axis, though it was probably a few cm/meters away from it in reality when it struck the telescope.\n",
```

Comment:
This is being super pedantic, but there are 2 dashed lines lines so it might be even more clear to specify the vertical dashed line. 

Also I think the "few cm/meters away" comment at the end might not be necessary and even a tad confusing.